build-logic/src/main/kotlin/org/terasology/gradology/ValidateJsonAssets.kt

```
@@ -0,0 +1,208 @@
+                }
+
+                walk(reader, file, "", warnings)
+
+                // Content after the root value is a defect, but not a fatal one: the engine's
+                // loaders read a single value and never check what follows, so the file still
+                // loads. Peeking can itself throw when the trailing bytes are not the start of a
+                // value (a stray closing brace, say), so that has to be caught here rather than
+                // by the outer handler - otherwise it would be reported as a parse failure.
+                val hasTrailingContent = try {
+                    reader.peek() != JsonToken.END_DOCUMENT
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)